cluster-test/test-orchestrator/assertion.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"reflect"
+	"time"
+)
+
+// An Assertion checks if a Value is equal or is contained by an interval.
+type Assertion struct {
+	Type AssertionType // can be is or in
+
+	// This is the Value of the Assertion in case of AssertionTypeIs
+	// or the first Value of the interval in case of AssertionTypeIn.
+	V1 Value
+
+	// The second Value of the interval in case of AssertionTypeIn.
+	// This value is ignored in case of AssertionTypeIs.
+	V2 Value
```

Comment:
Why would this struct ever need more params?

I can't think of anything we want to assert on that needs more values than an interval.